packages/next/src/server/future/route-matcher-providers/dev/dev-app-route-route-matcher-provider.ts

```diff
@@ -37,6 +38,7 @@ export class DevAppRouteRouteMatcherProvider extends FileCacheRouteMatcherProvid
         // The pathname to match should have the trailing `/route` and other route
         // group information stripped from it.
         wrapNormalizerFn(normalizeAppPath),
+        new UnderscoreNormalizer(),
```

Comment:
Are those two always together? Maybe we should combine them/export them as a set of normaliser for app router